src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/BaseOtlpHttpExportClient.cs

```diff
@@ -34,24 +36,33 @@ protected BaseOtlpHttpExportClient(OtlpExporterOptions options, HttpClient httpC
     internal IReadOnlyDictionary<string, string> Headers { get; }
 
     /// <inheritdoc/>
-    public bool SendExportRequest(TRequest request, CancellationToken cancellationToken = default)
+    public ExportClientResponse SendExportRequest(TRequest request, CancellationToken cancellationToken = default)
     {
+        DateTime deadline = DateTime.UtcNow.AddMilliseconds(this.HttpClient.Timeout.TotalMilliseconds);
```

Comment:
Why does this not use the timeout value from the OtlpExporterOptions?

Reply:
good point - It is because user can override it with their own HttpClient instance via [HttpClientFactory](https://github.com/open-telemetry/opentelemetry-dotnet/blob/e4b08ac152fe12e88a124f099c6d7853e4c505fc/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs#L180). So using the timeout value from the client itself and not options.

Reply:
Do we have a way to detect if the user actually configured timeout through OtlpExporter options?

Reply:
No - even if they are not using custom client. We initialize default client using timeout value set via options. https://github.com/open-telemetry/opentelemetry-dotnet/blob/e4b08ac152fe12e88a124f099c6d7853e4c505fc/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs#L82-L88

So in the both the cases we will have the correct deadline here.

Reply:
It would be good to add this as a comment.

Reply:
Sure, I can add it in my next PR if thats ok.